src/qibolab/platform.py

```diff
@@ -178,9 +216,30 @@ def execute_pulse_sequences(self, sequences: List[PulseSequence], options: Execu
             **kwargs: May need them for something
         Returns:
             Readout results acquired by after execution.
-
         """
-        return self._execute("play_sequences", sequences, options, **kwargs)
+        options = self.settings.apply(options)
+
+        duration = sum(seq.duration for seq in sequences)
+        time = (duration + len(sequences) * options.relaxation_time) * options.nshots * 1e-9
+        log.info(f"Minimal execution time (unrolling): {time}")
+
+        try:
+            return self._execute("play_sequences", sequences, options, **kwargs)
+        except NotImplementedError:
+            # find readout pulses
+            ro_pulses = {pulse.serial: pulse.qubit for sequence in sequences for pulse in sequence.ro_pulses}
+
+            results = defaultdict(list)
+            for batch in self.controller.split_batches(sequences):
+                sequence, readouts = unroll_sequences(batch, options.relaxation_time)
+                result = self._execute("play", sequence, options, **kwargs)
+                for serial, new_serials in readouts.items():
+                    results[serial].extend(result[s] for s in new_serials)
```

Comment:
```suggestion
                    results[serial].extend(result[ser] for ser in new_serials)
```

Just to avoid going crazy :)